### ISSUE TITLE
[scan][2.201.2] Option `should_zip_build_products` doesn't work anymore with Xcode 12.5.1

### ISSUE THREAD

**Issue by MartyCatawiki** [fastlane/fastlane#19865]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

We just just updated from fastlane version 2.199.0 to 2.201.2 but we noticed that the `scan` option `should_zip_build_products` is not working anymore for 2.201.2.
This is all run on CircleCI

##### Command executed

    scan(
      scheme: ENV["PROJECT_SCHEME_UI_TESTS"], # XCTest scheme
      clean: true,                          # Recommended: This would ensure the build would not include unnecessary files
      skip_slack: true,                     # No need to post scan result on slack
      skip_detect_devices: true,            # Required
      build_for_testing: true,              # Required
      sdk: "iphoneos",                      # Required
      should_zip_build_products: true       # Must be true to set the correct format for Firebase Test Lab
    )

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre> +------------------------------------------------+--------------------------------------------------------------------------------------------+
|                                                          Summary for scan 2.201.2                                                           |
+------------------------------------------------+--------------------------------------------------------------------------------------------+
| scheme                                         | CatawikiUITests                                                                            |
| clean                                          | true                                                                                       |
| skip_slack                                     | true                                                                                       |
| skip_detect_devices                            | true                                                                                       |
| build_for_testing                              | true                                                                                       |
| sdk                                            | iphoneos                                                                                   |
| should_zip_build_products                      | true                                                                                       |
| workspace                                      | ./Catawiki.xcworkspace                                                                     |
| ensure_devices_found                           | false                                                                                      |
| force_quit_simulator                           | false                                                                                      |
| reset_simulator                                | false                                                                                      |
| disable_slide_to_type                          | true                                                                                       |
| reinstall_app                                  | false                                                                                      |
| open_report                                    | false                                                                                      |
| output_directory                               | ./fastlane/test_output                                                                     |
| output_types                                   | html,junit                                                                                 |
| buildlog_path                                  | ~/Library/Logs/scan                                                                        |
| include_simulator_logs                         | false                                                                                      |
| xcodebuild_formatter                           | xcpretty                                                                                   |
| output_remove_retry_attempts                   | false                                                                                      |
| derived_data_path                              | /Users/distiller/Library/Developer/Xcode/DerivedData/Catawiki-dazbzqlvgzzsmsfbitxhtgnfeekb |
| output_xctestrun                               | false                                                                                      |
| result_bundle                                  | false                                                                                      |
| use_clang_report_name                          | false                                                                                      |
| disable_concurrent_testing                     | false                                                                                      |
| skip_build                                     | false                                                                                      |
| slack_use_webhook_configured_username_and_icon | false                                                                                      |
| slack_username                                 | fastlane                                                                                   |
| slack_icon_url                                 | https://fastlane.tools/assets/img/fastlane_icon.png                                        |
| slack_only_on_failure                          | false                                                                                      |
| xcodebuild_command                             | env NSUnbufferedIO=YES xcodebuild                                                          |
| skip_package_dependencies_resolution           | false                                                                                      |
| disable_package_automatic_updates              | false                                                                                      |
| use_system_scm                                 | false                                                                                      |
| number_of_retries                              | 0                                                                                          |
| fail_build                                     | true                                                                                       |
| xcode_path                                     | /Applications/Xcode-12.5.1.app                                                             |
+------------------------------------------------+--------------------------------------------------------------------------------------------+
 </pre>
</details>

### Missing part (copied from output 2.199.0 version

<details>
  <pre>[08:24:35]: ▸ Test build Succeeded
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 0 |
| Number of failures | 0 |
+--------------------+---+

[08:24:35]: Zipping build products
[08:24:35]: $ cd '/Users/distiller/Library/Developer/Xcode/DerivedData/Catawiki-dazbzqlvgzzsmsfbitxhtgnfeekb/Build/Products' && rm -f '/Users/distiller/project/fastlane/test_output/build_products.zip' && zip -r '/Users/distiller/project/fastlane/test_output/build_products.zip' *
[08:25:24]: Successfully zipped build products: /Users/distiller/project/fastlane/test_output/build_products.zip </pre>
</details>

**Comment by MartyCatawiki** [comment]
> It seems like you have not included the output of `fastlane env`
> To make it easier for us help you resolve this issue, please update the issue to include the output of `fastlane env` 👍

It's all running on CircleCI, but it is version 2.201.2

**Comment by crazymanish** [comment]
Tagging @joshdholtz in this issue. looks like the recent `xcbeautify` migration has lots of changes inside the `scan` tool.

**Comment by crazymanish** [comment]
Tagging @joshdholtz in this issue.
My wild guess is, it is broken because of `output_directory` changes inside Recent `xcbeautify` migration [PR](https://github.com/fastlane/fastlane/pull/19629)
https://github.com/fastlane/fastlane/blob/9b8ff3e70bca61abfaf4e50ecdff7e9456ba4cef/scan/lib/scan/runner.rb#L322

**Comment by Abdullah-TH** [comment]
I have a related scan issue started in version 2.201.2 when trying to test a swift package
<details>
  <summary>error output</summary>
  
```
/Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/scan/lib/scan/runner.rb:178:in `join': \e[31m[!] no implicit conversion of nil into String\e[0m (TypeError)
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/scan/lib/scan/runner.rb:178:in `find_xcresults_in_derived_data'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/scan/lib/scan/runner.rb:23:in `run'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/scan/lib/scan/manager.rb:23:in `work'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/actions/run_tests.rb:17:in `run'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	from Fastfile:188:in `block (2 levels) in parsing_binding'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/commands_generator.rb:109:in `block (2 levels) in run'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/commands_generator.rb:353:in `run'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/gems/fastlane-2.201.2/bin/fastlane:23:in `<top (required)>'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/bin/fastlane:25:in `load'
	from /Users/abdullahth/jaheziosv3/vendor/bundle/ruby/3.1.0/bin/fastlane:25:in `<top (required)>'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/cli/exec.rb:58:in `load'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/cli/exec.rb:23:in `run'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/cli.rb:484:in `exec'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/cli.rb:31:in `dispatch'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/cli.rb:25:in `start'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/exe/bundle:48:in `block in <top (required)>'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /Users/abdullahth/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bundler-2.3.6/exe/bundle:36:in `<top (required)>'
	from /Users/abdullahth/.rbenv/versions/3.1.0/bin/bundle:25:in `load'
	from /Users/abdullahth/.rbenv/versions/3.1.0/bin/bundle:25:in `<main>'
```
</details>

<details>
  <summary>Fastfile code that trigger the error</summary>
  
```
 run_tests(
      package_path:"App",
      sdk:"iphonesimulator",
      scheme:"App",
      destination:"platform=iOS Simulator,name=iPhone 13 Pro"
    )
```
</details>

**Comment by joshdholtz** [comment]
@Abdullah-TH Your issue should be fixed by #19871 😊 

@MartyCatawiki Would you be able to provide more info about your issues?  Is `scan` sill running successfully and just not making a zip with the correct contents? Is `scan` throwing an error when trying to zip? Full console output or seeing a stacktrace would be great!

**Comment by MartyCatawiki** [comment]
@joshdholtz Thanks for looking into this. The `scan` is running successfully but just not making a zip and also not showing an error or something. 
What is missing after normally the `Test build Succeeded` is this:
<pre>
[08:24:35]: ▸ Test build Succeeded
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 0 |
| Number of failures | 0 |
+--------------------+---+
[08:24:35]: Zipping build products

[08:24:35]: $ cd '/Users/distiller/Library/Developer/Xcode/DerivedData/Catawiki-dazbzqlvgzzsmsfbitxhtgnfeekb/Build/Products' && rm -f '/Users/distiller/project/fastlane/test_output/build_products.zip' && zip -r '/Users/distiller/project/fastlane/test_output/build_products.zip' *

[08:25:24]: Successfully zipped build products: /Users/distiller/project/fastlane/test_output/build_products.zip 
</pre>

Also another difference i noticed is at the end of 2.199.0
<pre>
| SCAN_ZIP_BUILD_PRODUCTS_PATH       | /Users/distiller/project/fastlane/test_output/build_products.zip                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
| SCAN_DERIVED_DATA_PATH             | /Users/distiller/Library/Developer/Xcode/DerivedData/Catawiki-dazbzqlvgzzsmsfbitxhtgnfeekb                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| SCAN_GENERATED_PLIST_FILES         | []                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
| SCAN_GENERATED_PLIST_FILE          |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
</pre>
Compared to 2.201.2
<pre>
| SCAN_GENERATED_XCRESULT_PATH       |                                                                                                                                                                                                                                                                                                              |
| SCAN_DERIVED_DATA_PATH             | /Users/distiller/Library/Developer/Xcode/DerivedData/Catawiki-dazbzqlvgzzsmsfbitxhtgnfeekb                                                                                                                                                                                                                   |
| SCAN_GENERATED_PLIST_FILES         | []                                                                                                                                                                                                                                                                                                           |
| SCAN_GENERATED_PLIST_FILE          |    
</pre>

**Comment by crazymanish** [comment]
Hi @MartyCatawiki 
Open [PR](https://github.com/fastlane/fastlane/pull/19875) to fix this issue. Can you please test once! 🙏 
Testing steps are there inside the PR description.

**Comment by MartyCatawiki** [comment]
Thanks a lot @crazymanish, this solves the issue 🙇🏼‍♂️ ❤️

**Comment by MartyCatawiki** [comment]
> The pull request #19875 that closed this issue was merged and released as part of [_fastlane_ 2.204.0](https://github.com/fastlane/fastlane/releases/tag/2.204.0) 🚀
> Please let us know if the functionality works as expected as a reply here. If it does not, please open a new issue. Thanks!

Yes it works now 🥳 , thanks a lot 🙇🏼‍♂️
